pkg/api/v1alpha1/imagemap_types.go

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2019-Present Pivotal Software, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: json tags are required.  Any new fields must have json tags for the fields to be serialized.
+
+// ImageMapSpec defines the desired state of ImageMap
+type ImageMapSpec struct {
+	Map map[string]string `json:"map,omitempty"`
+}
+
+func (in *ImageMapSpec) DeepCopyInto(out *ImageMapSpec) {
```

Comment:
DeepCopy methods are typically generated and put into a separate file (zz_generated.deepcopy.go) so they can be updated without impacting the main source.

Reply:
I put the generated methods in such a file but had to add these manually.